direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -411,16 +411,20 @@ private void handleFlushDataBaseOnPosition(
 
       Runnable onIdle =
           () -> {
-            synchronized (observer) {
-              watermark.idle();
-              observer.onIdle(watermark::getWatermark);
-              if (watermark.getWatermark()
-                  >= (Watermarks.MAX_WATERMARK - BOUNDED_OUT_OF_ORDERNESS)) {
-                observer.onCompleted();
-                getObservers(getUri()).remove(consumerId);
-                onIdleRef.get().cancel(true);
-                killSwitch.set(true);
+            try {
+              synchronized (observer) {
+                watermark.idle();
+                observer.onIdle(watermark::getWatermark);
+                if (watermark.getWatermark()
+                    >= (Watermarks.MAX_WATERMARK - BOUNDED_OUT_OF_ORDERNESS)) {
+                  observer.onCompleted();
+                  getObservers(getUri()).remove(consumerId);
+                  onIdleRef.get().cancel(true);
+                  killSwitch.set(true);
+                }
               }
+            } catch (Throwable ex) {
+              log.warn("Exception in onIdle", ex);
```

Comment:
are you sure about this?

Reply:
What's wrong with that?

Reply:
If exception is not caught, the `onIdle` mechanism breaks.

Reply:
that we are basically ignoring the exception here, should this be handled somehow?

Reply:
Not sure how. Any suggestions?